ui/components/src/ui-components/common/ConditionalAutoSizer.tsx

```diff
@@ -0,0 +1,42 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import * as React from "react";
+import AutoSizer, { Size } from "react-virtualized-auto-sizer";
+
+/** @internal */
+export interface ConditionalAutoSizerProps {
+  width?: number;
+  height?: number;
+  onResize?: (newSize: Size) => void;
+  children: (size: Size) => React.ReactNode;
+}
+
+/** @internal */
+export function ConditionalAutoSizer(props: ConditionalAutoSizerProps): React.ReactElement {
```

Comment:
Even thought this is internal, could you add comment to this function describing its use?

Reply:
Added some documentation in 30fd7b8e93fc04b93c9f9aea37caf789a9a2518e.

Reply:
Thanks!